File: practice/algorithms/implementation/jumping_on_the_clouds_revisited/JumpingOnTheCloudsRevisited.scala
object JumpingOnTheCloudsRevisited extends App {
  val lines = Source.stdin.getLines().toList
  val nk = lines.head.split(" ").map(_.toInt)
  val n = nk(0)
  val k = nk(1)
  val clouds = lines(1).split(" ").map(_.toInt)
  var energy = 100
  var pos = k % n
  energy -= (if(clouds(pos) == 1) 3 else 1)
  while(pos != 0) {
    pos = (pos + k) % n
    energy -= (if(clouds(pos) == 1) 3 else 1)
  }
  println(energy)
}

File: practice/algorithms/implementation/jumping_on_the_clouds_revisited/jumping_on_the_clouds_revisited.rb
(n, k) = gets.split(" ").map(&:to_i)
clouds = gets.split(" ").map(&:to_i)

energy = 100
pos = k % n
energy -= ((clouds[pos] == 1) ? 3 : 1)
while(pos != 0) do
  pos = (pos + k) % n
  energy -= ((clouds[pos] == 1) ? 3 : 1)
end
puts energy